File: PHP/7php 2/maxMultiple.php
<?php
/*
|--------------------------------------------------------------------------
| Maximum Multiple
|--------------------------------------------------------------------------
|
*/

function maxMultiple($divisor, $bound) {
   return $bound - $bound % $divisor;
}


function maxMultiple($divisor, $bound) {
   while ($bound % $divisor != 0) {
      $bound--;
   }
   return $bound;
}


function maxMultiple($divisor, $bound) {
   return intdiv($bound, $divisor) * $divisor;
}


print_r(maxMultiple(2, 7) . ' ' . 6 . '
');
print_r(maxMultiple(3, 10) . ' ' . 9 . '
');
print_r(maxMultiple(7, 17) . ' ' . 14 . '
');

File: Python/7py 2/max_multiple.py
"""
|--------------------------------------------------------------------------
| Maximum Multiple
|--------------------------------------------------------------------------
|
"""

def max_multiple(divisor, bound):
   return bound - bound % divisor


def max_multiple(divisor, bound):
   return bound // divisor * divisor


def max_multiple(divisor, bound):
   return max([n for n in range(1, bound + 1) if n % divisor == 0])


print(max_multiple(2,7),6)
print(max_multiple(3,10),9)
print(max_multiple(7,17),14)